app/javascript/react/components/CoffeeShopShowContainer.js

```diff
@@ -3,20 +3,30 @@ import ReviewsContainer from './ReviewsContainer';
 import ReviewFormContainer from './ReviewFormContainer';
 
 const CoffeeShopShowContainer = (props) => {
-  const [coffeeShop, setCoffeeShop] = useState({ reviews: [] });
+  const [coffeeShop, setCoffeeShop] = useState({});
+  const [reviews, setReviews] = useState([]);
+  const [currentUser, setCurrentUser] = useState({});
   const [errors, setErrors] = useState({});
-  let currentUser = coffeeShop.current_user;
   let coffeeShopId = props.match.params.id;
 
   const fetchCoffeeShop = async () => {
     try {
       const coffeeShopResponse = await fetch(`/api/v1/coffee_shops/${coffeeShopId}`);
       if (coffeeShopResponse.ok) {
         const parsedCoffeeShopResponse = await coffeeShopResponse.json();
-        return setCoffeeShop(parsedCoffeeShopResponse.coffee_shop);
+        const coffeeShop = ({ name, description, image_url, address, city, state, zip }) => ({
+          name,
+          description,
+          image_url,
+          address,
+          city,
+          state,
+          zip
+        });
+        setCoffeeShop(coffeeShop(parsedCoffeeShopResponse.coffee_shop));
```

Comment:
Im...struggling to understand why you need all of this destructuring? I see you use it on 26, but it doesnt...do anything? why not just have

` setCoffeeShop(parsedCoffeeShopResponse.coffee_shop);`

otherwise, your backend serializer should return data in the format that you want such that it is easy to add to state